### PR TITLE
(Support) Export as Image

### PR DESCRIPTION
**Pull request by dcupidon**
Exposed a viewer and a standalone method to convert the pages of a PDF to JPEG, BMP or PNG

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -2918,15 +2947,17 @@ this._viewer.canRedo().then((canRedo) => {
 ### Others
 
 #### exportAsImage
-Export a PDF page to image format defined in `Config.ExportFormat`.
+Export a PDF page to an image format defined in [`Config.ExportFormat`](./src/Config/Config.js). 
+
+Unlike RNPdftron.exportAsImage, this is a viewer method and should only be called *after* a `DocumentView` instance has been created or else unexpected behaviour can occur. This method uses the PDF that is associated with the viewer, and does not take a local file path to the desired PDF.
```
Comment (sgong-pdftron) [inline API.md:2952]: - [ ] after document loaded

File: android/src/main/java/com/pdftron/reactnative/modules/RNPdftronModule.java [hunk 2]
```diff
@@ -151,4 +153,15 @@ public void pdfFromOfficeTemplate(final String docxPath, final ReadableMap json,
             promise.reject(e);
         }
     }
+
+    @ReactMethod
+    public void exportAsImage(int pageNumber, double dpi, String exportFormat, final String filePath, final Promise promise) {
+        try {
+            PDFDoc doc = new PDFDoc(filePath);
+            String imagePath = ReactUtils.exportAsImageHelper(doc, pageNumber, dpi, exportFormat);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/modules/RNPdftronModule.java:161]: - [ ] read lock on pdfdoc

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Review by sgong-pdftron — APPROVED** [review]

File: ios/RNPdftron.m [hunk 3]
```diff
@@ -142,6 +167,36 @@ - (NSError *)errorFromException:(NSException *)exception
...
+        shouldUnlock = YES;
+
+        PTPDFDraw *draw = [[PTPDFDraw alloc] initWithDpi:dpi];
+        NSString* tempDir = NSTemporaryDirectory();
+        NSString* fileName = [NSUUID UUID].UUIDString;
+        resultImagePath = [tempDir stringByAppendingPathComponent:fileName];
+        resultImagePath = [resultImagePath stringByAppendingPathExtension:exportFormat];
+        PTObjSet* hintSet = [[PTObjSet alloc] init];
+        PTObj* encoderHints = [hintSet CreateArray];
+        [encoderHints PushBackName:@"JPEG"];
```
Comment (JamesTron) [inline ios/RNPdftron.m:185]: This code seems to be ignoring the requested `exportFormat` and always exporting as JPEG.

File: ios/RNTPTDocumentViewModule.m [hunk 4]
```diff
@@ -85,7 +85,7 @@ - (NSError *)errorFromException:(NSException *)exception
                  rejecter:(RCTPromiseRejectBlock)reject)
 {
     @try {
-        NSString *path = [[self documentViewManager] exportAsImageForDocumentViewTag:tag pageNumber:pageNumber dpi:dpi imageFormat:(NSString*)imageFormat];
+        NSString *path = [[self documentViewManager] exportAsImageForDocumentViewTag:tag pageNumber:pageNumber dpi:dpi imageFormat:imageFormat];
```
Comment (JamesTron) [inline ios/RNTPTDocumentViewModule.m:88]: Everywhere else the param `imageFormat` in this method is referred to as `exportFormat`. Should this one be `exportFormat` too, either for correctness or consistency?

**Review by JamesTron — CHANGES_REQUESTED** [review]

**Review by sgong-pdftron — APPROVED** [review]

**Review by JamesTron — APPROVED** [review]

**Pull request merged**